CodeEditModules/Modules/Keybindings/src/CommandManager.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  CommandManager.swift
+//
+//  Created by Alex on 23.05.2022.
+//
+
+import Foundation
+import WorkspaceClient
+
+/// Usage
+///
+/**
+```
+let mgr = CommandManager.init()
+let wrap = ClosureWrapper.init(closure: {
+    print("testing closure")
+})
+mgr.addCommand(name: "test", command: wrap)
+mgr.executeCommand(name: "test")
+ ```
+ */
+
+public final class CommandManager: ObservableObject {
+    @Published private var commandsList: [String: Command]
+
+    private init() {
+        commandsList = [:]
+    }
+
+    public static let shared: CommandManager = .init()
+
+    public func addCommand(name: String, title: String, id: String, command: ClosureWrapper) {
+        let command = Command.init(id: name, title: title, closureWrapper: command)
+        commandsList[id] = command
+    }
+
+    public var commands: [Command] {
+        return commandsList.map { $0.value }
+    }
+
+    public func executeCommand(name: String) {
+        commandsList[name]?.closureWrapper.call()
+    }
+}
+
+public struct Command: Identifiable, Hashable {
```

Comment:
Please add proper documentation to `public` classes, structs, methods and variables/constants.

A guide about code documentation can be found [here](https://nshipster.com/swift-documentation/).